### DISCUSSION TITLE
Authentication issue with Claude API, Error 401

### DISCUSSION THREAD

**Discussion by mrussogit** [danny-avila/LibreChat#3996] [Troubleshooting] [answered]
Hey guys. 
When I installed LibreChat and started using it all went smooth. The next day it gave me error of what looks like invalid API key. But the API is valid and active from Anthropic Console. I tried refreshing the browser cache, creating another API key.. nothing works. 

`Something went wrong. Here's the specific error message we encountered: Operation failed after 3 attempts: 401 {"type":"error","error":{"type":"authentication_error","message":"invalid x-api-key"}}`

Any idea is greatly appreciated. LibreChat is **SO MUCH BETTER** than Anthropic Console.

**Comment by danny-avila** [comment]
There's no issue here: 

![image](https://github.com/user-attachments/assets/e4b1ed46-3975-4962-bf9e-5e686af051ce)


have you already tried a new API key?

**Reply by mrussogit** [reply]
I created a new key, now it works. That's odd. I'll the other key is also valid. I wonder if I'll need to update a new key every day. I checked to never expire and the key is set to never expire. I'll see how it goes the next few days.. thanks for now!

**Reply by mrussogit** [reply]
I haven't closed the tab with LibreChat I was using ChatGPT afternoon. First I got 
![image](https://github.com/user-attachments/assets/846435fd-c4f0-41a1-aac3-13413b681118)

then
![image](https://github.com/user-attachments/assets/c58a72cc-96b1-4ec2-8e7b-fd064b7f1d91)

**Reply by danny-avila** [reply, marked as answer]
Are you setting the key from the frontend UI? There could be an issue there. 


For now, try setting it from ENV file instead

https://www.librechat.ai/docs/configuration/pre_configured_ai/anthropic

**Reply by mrussogit** [reply]
> Are you setting the key from the frontend UI? There could be an issue there.
> 
> For now, try setting it from ENV file instead
> 
> https://www.librechat.ai/docs/configuration/pre_configured_ai/anthropic

Yes, from frontend UI. Now I did set the API Key from .env it didn't work. I placed the key from frontend and it worked. Closed Librechat Restarted LibreChat on docker, it worked in first try. Possibly a glitch in API key from frontend. I'll keep poking it in the next 1-2 days and see if I can reproduce before marking as answer. Many thanks!

**Reply by mrussogit** [reply]
After launching docker, with the api key saved in .env looks like it works! Thanks!
